### PR TITLE
chore(profiling): update ddup to libdatadog5, add better error handling

### PR DESCRIPTION
**Pull request by sanchda**
Updates libdatadog version to 5, adds some error checking.

## Checklist

- [X] Change(s) are motivated and described in the PR description.
- [X] Testing strategy is described if automated tests are not included in the PR.
- [X] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [X] Change is maintainable (easy to change, telemetry, documentation).
- [X] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [X] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [X] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [X] Title is accurate.
- [X] No unnecessary changes are introduced.
- [X] Description motivates each change.
- [X] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [X] Testing strategy adequately addresses listed risk(s).
- [X] Change is maintainable (easy to change, telemetry, documentation).
- [X] Release note makes sense to a user of the library.
- [X] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [X] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [X] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [X] This PR doesn't touch any of that.

### CODE REVIEW

**Review by emmettbutler — APPROVED** [review]
The python bits look good to me!

File: ddtrace/internal/datadog/profiling/src/interface.cpp [hunk 1]
```diff
@@ -127,7 +126,14 @@ sigsegv_handler(int sig, siginfo_t* si, void* uc)
 }
 
 #endif
-void
+char* ddup_errmsg = nullptr;
```
[the remark this answers is not in the document]
Comment (danielsn) [reply ddtrace/internal/datadog/profiling/src/interface.cpp:129]: What happens if there are multiple threads? Are these thread local storage?

[the remark this answers is not in the document]
Comment (sanchda) [reply ddtrace/internal/datadog/profiling/src/interface.cpp:129]: As discussed, will make a few fixups in a subsequent PR
* At initialization, ddup will acquire the logger module
* During error, ddup interfaces will push errors directly to the logger
* ddup will return state (true/false or an enum if appropriate) to the caller, but not furnish direct logs (those go into the logger)
* Logger wrapper will probably have to take the GIL

**Review by danielsn — APPROVED** [review]
Approved given the upcoming fixup for errno.  Recommend adding a comment about the magic constant `0.51` to help your future self.

File: ddtrace/profiling/collector/memalloc.py [hunk 2]
```diff
@@ -171,7 +170,7 @@ def collect(self):
                 if thread_id in thread_id_ignore_set:
                     continue
                 ddup.start_sample(nframes)
-                ddup.push_alloc(int((ceil(size) * alloc_count) / count), count)  # Roundup to help float precision
+                ddup.push_alloc(int(((size + 0.51) * alloc_count) / count), count)  # Roundup to help float precision
```
[the remark this answers is not in the document]
Comment (sanchda) [reply ddtrace/profiling/collector/memalloc.py:173]: Will add a comment clarifying the change

**Comment by sanchda** [comment]
This PR has been made pointless by an incoming stack of PRs, so in the interest of keeping the git history from being totally inane, I'm deleting this PR.  The reviews here have not been pointless and will be rolled into the other work.  Thanks everyone!

**Pull request closed without merging**